searchlib/src/vespa/searchlib/features/dotproductfeature.cpp

```diff
@@ -621,9 +633,26 @@ DotProductBlueprint::prepareSharedState(const IQueryEnvironment & env, IObjectSt
         {
             attribute = env.getAttributeContext().getAttributeStableEnum(getAttribute(env));
         }
+        fef::Anything::UP arguments;
+        if (attribute->getCollectionType() == attribute::CollectionType::ARRAY) {
+            Property tensorBlob = env.getProperties().lookup(getBaseName(), _queryVector, "tensor");
+            if (attribute->isFloatingPointType() && tensorBlob.found() && !tensorBlob.get().empty()) {
+                const Property::Value & blob = tensorBlob.get();
+                vespalib::nbostream stream(blob.data(), blob.size());
+                if (attribute->getBasicType() == BasicType::FLOAT) {
+                    arguments = std::make_unique<ArrayParam<float>>(stream);
+                } else {
+                    arguments = std::make_unique<ArrayParam<double>>(stream);
+                }
```

Comment:
here we have successfully extracted the tensor value, but we nevertheless also look up the non-tensor parameter property further down without using it.

Reply:
Good catch, that was not intentional. Fixed now.